hist/hist/src/TH2.cxx

```diff
@@ -1550,37 +1580,94 @@ TH2 *TH2::Rebin2D(Int_t nxgroup, Int_t nygroup, const char *newname)
       Error("Rebin2D", "Illegal value of nygroup=%d",nygroup);
       return 0;
    }
+   if (!newname && xbins) {
+       Error("Rebin","if xbins is specified, newname must be given");
+       return 0;
+    }
+   if (!newname && ybins) {
+       Error("Rebin","if ybins is specified, newname must be given");
+       return 0;
+    }
 
    Int_t newxbins = nxbins / nxgroup;
-   Int_t newybins = nybins / nygroup;
    Int_t newnx = newxbins + 2; // regular bins + overflow / underflow
+   if (!xbins) {
+      Int_t nbgx = nxbins/nxgroup;
+      if (nbgx*nxgroup != nxbins) {
+         Warning("Rebin2D", "nxgroup=%d is not an exact divider of nxbins=%d.",nxgroup,nxbins);
+      }
+   }
+   else {
+   // in the case that xbins is given (rebinning in variable bins), nxgroup is
+   // the new number of bins and number of grouped bins is not constant.
+   // when looping for setting the contents for the new histogram we
+   // need to loop on all bins of original histogram. Then set nxgroup=nxbins
+      newxbins = nxgroup;
+      nxgroup = nxbins;
+   }
+
+   Int_t newybins = nybins / nygroup;
    Int_t newny = newybins + 2; // regular bins + overflow / underflow
+   if (!ybins) {
+      Int_t nbgy = nybins/nygroup;
+      if (nbgy*nygroup != nybins) {
+         Warning("Rebin2D", "nygroup=%d is not an exact divider of nybins=%d.",nygroup,nybins);
+      }
+   }
+   else {
+   // in the case that ybins is given (rebinning in variable bins), nygroup is
+   // the new number of bins and number of grouped bins is not constant.
+   // when looping for setting the contents for the new histogram we
+   // need to loop on all bins of original histogram. Then set nygroup=nybins
+      newybins = nygroup;
+      nygroup = nybins;
+   }
 
    // Save old bin contents into a new array
+   Double_t entries = fEntries;
    Double_t *oldBins = new Double_t[fNcells];
    for (Int_t i = 0; i < fNcells; ++i) oldBins[i] = RetrieveBinContent(i);
 
    Double_t* oldErrors = NULL;
-   if (fSumw2.fN) {
+   if (fSumw2.fN != 0) {
```

Comment:
Is it a pointer? (That's hard to see from the review snippet on github)
If yes, put `!= nullptr`.

Reply:
no `fN` is `Int_t`